Cargo.toml

```diff
@@ -14,29 +14,30 @@ exclude = ["rust-toolchain", ".vscode"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-bytes = "0.5.1"
-futures = "0.3.1"
-log = "0.4.8"
-nom = "5.0.1"
-owning_ref = "0.4.0"
-pin-project = "0.4.17"
-rand = "0.7.2"
-serde = { version = "1.0.101", features = ["derive"] }
-serde_json = "1.0.40"
-tokio = { version = "0.2.1", features = ["dns", "io-util", "macros", "stream", "sync", "rt-core", "time", "tcp"] }
-tokio-util = { version = "0.2.0", features = ["codec"] }
-uuid = { version = "0.7.4", features = ["v4"] }
+bytes = "*"
+futures = "*"
+log = "*"
+nom = "*"
+owning_ref = "*"
+pin-project = "*"
+rand = "*"
+serde = { version = "*", features = ["derive"] }
+serde_json = "*"
+tokio = { version = "*", features = ["io-util", "macros", "sync", "time", "net"] }
+tokio-util = { version = "*", features = ["codec"] }
+tokio-stream = { version = "*", features = ["sync"] }
+uuid = { version = "*", features = ["v4"] }
```

Comment:
Hey Matt, this is awesome! Thanks for the PR.

The one change is I generally think it is bad form to have `"*"` dependencies in a library. Could you change the dependency sections to

```toml
[dependencies]
bytes = "1.0"
futures = "0.3"
log = "0.4"
nom = "6.2"
owning_ref = "0.4"
pin-project = "1.0"
rand = "0.8"
serde = { version = "1.0", features = ["derive"] }
serde_json = "1.0"
tokio = { version = "1.9", features = ["io-util", "macros", "sync", "time", "net"] }
tokio-util = { version = "0.6", features = ["codec"] }
tokio-stream = { version = "0.1", features = ["sync"] }
uuid = { version = "0.8", features = ["v4"] }

# Optional dependencies
native-tls-crate = { version = "0.2", optional = true, package = "native-tls" }
rustls = { version = "0.19", optional = true }
tokio-native-tls = { version = "0.3", optional = true }
tokio-rustls = { version = "0.22", optional = true }

[dev-dependencies]
env_logger = "0.9"
hostname = "0.3"
tokio = { version = "1.9", features = ["process", "rt-multi-thread"] }
```